plugins/gitlab-catalog-backend/src/GitlabUserProcessor.ts

```diff
@@ -0,0 +1,150 @@
+import { LocationSpec } from '@backstage/plugin-catalog-common';
+import type {
+  CatalogProcessor,
+  CatalogProcessorEmit,
+} from '@backstage/plugin-catalog-node';
+import { ExpandedUserSchema, Gitlab } from '@gitbeaker/rest';
+import { Logger } from 'winston';
+
+const GITLAB_PER_PAGE_LIMIT = 500;
+const GITLAB_DEFAULT_HOST = 'https://gitlab.com';
+
+/**
+ * The GitlabUserProcessor is used to enrich our User entities with information
+ * from Gitlab: notably the user's Gitlab ID.
+ *
+ * @public
+ */
+export class GitlabUserProcessor implements CatalogProcessor {
+  // @ts-ignore: intended to reference as such by the library
+  private readonly gitlab: Gitlab;
+  private readonly logger: Logger;
+  private cacheLoaded: boolean;
+  private userLookup: Map<string, ExpandedUserSchema>;
+  // guarantee that users are loaded only once
+  private loadUserPromise: Promise<Map<string, ExpandedUserSchema>> | null =
+    null;
+
+  private async fetchUsers(): Promise<Map<string, ExpandedUserSchema>> {
+    if (!this.gitlab) {
+      return new Map();
+    }
+
+    if (!this.cacheLoaded) {
+      // we use a shared promise to make sure that the load is only
+      // executed once even if `GitlabUserProcessor.postProcessEntity`
+      // is called "concurrently".
+      if (!this.loadUserPromise) {
+        this.loadUserPromise = this.loadUsers();
+      }
+
+      // Wait for the data to load
+      await this.loadUserPromise;
+    }
+
+    return this.userLookup;
+  }
+
+  private async loadUsers(): Promise<Map<string, ExpandedUserSchema>> {
+    this.logger.info('Loading gitlab users');
+
+    let members: ExpandedUserSchema[] = [];
+    try {
+      members = await this.gitlab!.Users.all({
+        perPage: GITLAB_PER_PAGE_LIMIT,
+        active: true,
+        withoutProjectBots: true,
+      });
+    } catch (error) {
+      this.logger.error(`Error loading gitlab users: ${error}`);
+      return this.userLookup;
+    }
+
+    members.forEach(user => {
+      if (user.email) {
+        this.userLookup.set(user.email, user);
+      }
+    });
+
+    this.logger.info(`Loaded ${this.userLookup.size} gitlab users`);
+    this.cacheLoaded = true;
+
+    return this.userLookup;
+  }
+
+  static fromConfig(config: Config, logger: Logger): GitlabUserProcessor[] {
+    const token = config.getOptionalString('gitlabCatalog.token');
+    if (!token) {
+      logger.warn(
+        'No token provided for GitlabUserProcessor, skipping Gitlab user lookup',
+      );
+      return [];
+    }
+    let host = config.getOptionalString('gitlabCatalog.host');
+    if (!host) {
+      logger.info(
+        `No host provided for GitlabUserProcessor, defaulting to ${GITLAB_DEFAULT_HOST}`,
+      );
+      host = GITLAB_DEFAULT_HOST;
+    }
+    return [
+      new GitlabUserProcessor(
+        new Gitlab({
+          host: host,
+          token: token,
+        }),
+        logger,
+      ),
+    ];
+  }
+
+  // @ts-ignore: intended to reference as such by the library
```

Comment:
this is still throwing errors on tsc lint, any better ideas on how to ignore this line for now? @zhammer @colinodell 

bit out of our control since I'm referencing the `Gitlab` as the library authors intended it to be used

Reply:
hm let me try to dig in here, i'll pull down the code locally

Reply:
solution's here, i'll leave a comment with the suggestion below https://github.com/jdalrymple/gitbeaker/issues/2815#issuecomment-1311098696

Reply:
awesome thanks